src/Foundatio.Parsers.SqlQueries/SqlQueryParser.cs

```diff
@@ -0,0 +1,183 @@
+
+        return await GenerateSqlVisitor.RunAsync(node, context);
+    }
+
+    public SqlQueryVisitorContext GetContext(IEntityType entityType)
+    {
+        if (!_entityFieldCache.TryGetValue(entityType, out var fields))
+        {
+            fields = new List<EntityFieldInfo>();
+            AddEntityFields(fields, entityType);
+            _entityFieldCache.TryAdd(entityType, fields);
+        }
+
+        // make copy of fields list to avoid modifying the cached list
+        fields = fields.ToList();
+
+        var validationOptions = new QueryValidationOptions();
+        foreach (string field in fields.Select(f => f.Field))
+            validationOptions.AllowedFields.Add(field);
+
+        Configuration.SetValidationOptions(validationOptions);
+        return new SqlQueryVisitorContext
+        {
+            Fields = fields,
+            ValidationOptions = validationOptions
+        };
+    }
+
+    private void AddEntityFields(List<EntityFieldInfo> fields, IEntityType entityType, Stack<IEntityType> entityTypeStack = null, string prefix = null, bool isCollection = false, int depth = 0)
+    {
+        entityTypeStack ??= new Stack<IEntityType>();
+
+        if (depth > 0 && entityTypeStack.Contains(entityType))
+            return;
+
+        entityTypeStack.Push(entityType);
+
+        if (depth > Configuration.MaxFieldDepth)
+            return;
+
+        prefix ??= "";
+
+        foreach (var property in entityType.GetProperties())
+        {
+            string propertyPath = prefix + property.Name;
+            if (!Configuration.EntityTypePropertyFilter(property))
+                continue;
+
+            fields.Add(new EntityFieldInfo
+            {
+                Field = propertyPath,
+                IsNumber = property.ClrType.UnwrapNullable().IsNumeric(),
+                IsDate = property.ClrType.UnwrapNullable().IsDateTime(),
+                IsBoolean = property.ClrType.UnwrapNullable().IsBoolean(),
+                IsCollection = isCollection
+            });
+        }
+
+        foreach (var nav in entityType.GetNavigations())
+        {
+            string propertyPath = prefix + nav.Name;
+            if (!Configuration.EntityTypeNavigationFilter(nav))
+                continue;
+
+            AddEntityFields(fields, nav.TargetEntityType, entityTypeStack, propertyPath + ".", false, depth + 1);
+        }
+
+        foreach (var skipNav in entityType.GetSkipNavigations())
+        {
+            string propertyPath = prefix + skipNav.Name;
+            if (!Configuration.EntityTypeSkipNavigationFilter(skipNav))
+                continue;
+
+            AddEntityFields(fields, skipNav.TargetEntityType, entityTypeStack, propertyPath + ".", true, depth + 1);
+        }
+
+        entityTypeStack.Pop();
+    }
+
+    private void SetupQueryVisitorContextDefaults(IQueryVisitorContext context)
+    {
+        if (!context.Data.ContainsKey("@OriginalContextResolver"))
```

Comment:
Would it be worth having known consts for this?

Reply:
This is the same as the other code in this project. Don't want to change it now.